src/neighborsearch/neighborhood.jl

```diff
@@ -17,7 +17,11 @@ struct NeighborhoodSearch{O,N,T} <: NeighborSearchMethod
 end
 
 function NeighborhoodSearch(object::O, neigh::N) where {O,N}
-  tree = if neigh isa BallNeighborhood
+  if neigh isa EllipsoidNeighborhood
+    N1, N2 = ncoords(object), size(metric(neigh).qmat,1)
+    @assert  N1 == N2  "data and ellipse/ellipsoid must have the same dimensions"
+ end
```

Comment:
Do we really need this assertion here? If we want to add it, I think we need to first provide  public api to access the dimension of the ellipsoid neighborhood. The `metric(neigh).qmat` part is an internal detail.